pacsman/dcmtk_client.py

```diff
@@ -369,37 +372,40 @@ def fetch_thumbnail(self, study_id: str, series_id: str) -> Optional[str]:
                 image_ids.append(dataset.SOPInstanceUID)
 
         if not image_ids:
+            logger.error(f'Failed to find any image instances for series {series_id}')
             return None
 
         # try to get the middle image in the series for the thumbnail:
         #  instance ID order is usually the same as slice order but not guaranteed
         #  by the standard.
         middle_image_id = image_ids[len(image_ids) // 2]
+        return self._fetch_slice_thumbnail(study_id, series_id, middle_image_id)
+
+    def fetch_slice_thumbnail(self, study_id: str, series_id: str,
+                              instance_id: str) -> Optional[str]:
+        return self._fetch_slice_thumbnail(study_id, series_id, instance_id)
+
+    def _fetch_slice_thumbnail(self, study_id: str, series_id: str,
+                              instance_id: str) -> Optional[str]:
```

Comment:
Is there a reason `_fetch_slice_thumbnail` and `fetch_slice_thumbnail` are both defined? They look identical.

Reply:
`_fetch_slice_thumbnail` is the shared code between `fetch_thumbnail` and `fetch_slice_thumbnail`. I think it just needs a better name.

Reply:
Ahh, okay-- I see now. Thanks for clarifying